pandas/tests/arrays/test_datetimes.py

```diff
@@ -0,0 +1,96 @@
+"""
+Tests for DatetimeArray
+"""
+import pytest
+import numpy as np
+
+from pandas.core.arrays import DatetimeArrayMixin as DatetimeArray
+
+import pandas as pd
+import pandas.util.testing as tm
+
+
+class TestDatetimeArrayConstructors(object):
+
+    def test_scalar_raises_type_error(self):
+        # GH#23493
+        with pytest.raises(TypeError):
+            DatetimeArray(2)
+
+        with pytest.raises(TypeError):
+            pd.DatetimeIndex(pd.Timestamp.now())
+
+    def test_from_sequence_requires_1dim(self):
```

Comment:
This can be a test in the EA base tests. 

Although, thinking about it now, I am not sure we should require implementors to handle this case, as the method should never be called with 2D data to begin with. 

What is the reason you added handling of this to that method?

Reply:
> What is the reason you added handling of this to that method?

I'm not sure I understand the question.  Is there a reason not to?

Reply:
Is there ever a case where call `_from_sequence` internally with 2D data? And if so, what case? 

I would think that at the point we internally call `_from_sequence`, we are sure it is a 1D array (eg as the result of some operation).

Reply:
You're far more competent than I am and can be counted on not to make this particular mistake; I cannot.

If we're not allowing object-dtype/lists in `__init__/__new__` then we are basically forcing users to use the (private!) `_from_sequence`; validation need to happen somewhere.

Reply:
The `_from_sequence` method has a specific purpose in the EA interface: converting an array-like of scalars back to a proper Array (and should never be called by a user). Eg this method is used when doing an `astype(EAdtype)`, but I was thinking that it might well be that in those cases that we call it, it already has been checked the data is 1D. 

So I am simply honestly wondering if there is a specific case that you encountered now where this check was needed. 

But I suppose the answer is that you are using `_from_sequence` to validate generic user input, and there the user *can* pass 2D data and we should invalidate it.

However, it is not only you that do it, we actually already do it when doing `Series(..., dtype=EAdtype)` (where also user input is processed in `_from_sequence`), so indeed, we should probably have this check in general. 

So long story short: it indeed might make sense to add this type check. But then to repeat my initial comment: this should then be a test in the base EA tests to ensure all EAs do this properly?